pkg/core/transactor/listener.go

```diff
@@ -119,6 +124,47 @@ func (t *Transactor) handleAddress(r rpcbus.Request) error {
 	return nil
 }
 
+func (t *Transactor) handleGetTxHistory(r rpcbus.Request) error {
+	if t.w == nil {
+		return errWalletNotLoaded
+	}
+
+	records, err := t.w.FetchTxHistory()
+	if err != nil {
+		return err
+	}
+
+	if len(records) == 0 {
+		r.RespChan <- rpcbus.Response{*bytes.NewBufferString("No records found."), nil}
+		return nil
+	}
+
+	s := &strings.Builder{}
+	for _, record := range records {
+		// Direction
+		if record.Direction == txrecords.In {
+			s.WriteString("IN / ")
+		} else {
+			s.WriteString("OUT / ")
+		}
+		// Height
+		s.WriteString(strconv.FormatUint(record.Height, 10) + " / ")
+		// Time
+		s.WriteString(time.Unix(record.Timestamp, 0).Format(time.UnixDate) + " / ")
+		// Amount
+		s.WriteString(fmt.Sprintf("%.8f DUSK ", float64(record.Amount)/float64(wallet.DUSK)) + " / ")
+		// Unlock height
+		s.WriteString("Unlocks at " + strconv.FormatUint(record.UnlockHeight, 10) + " / ")
+		// Recipient
+		s.WriteString("Recipient: " + record.Recipient)
+
+		s.WriteString("\n")
+	}
+
+	r.RespChan <- rpcbus.Response{*bytes.NewBufferString(s.String()), nil}
```

Comment:
For now, this is OK. However, looking at the whole file, I suspect that tossing strings when returning values on the RPC bus is a code smell. I'd rather serialize a specific struct with a `String()` method. This way, if we'd ever need to migrate away from strings we would avoid the pain of backward incompatible behaviors

Reply:
Good point. I will put it in a separate issue.